titus-server-federation/src/main/java/com/netflix/titus/federation/service/CellWebClientConnectorUtil.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.federation.service;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+import com.netflix.titus.common.util.CollectionsExt;
+import com.netflix.titus.common.util.tuple.Either;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class CellWebClientConnectorUtil {
+
+    /**
+     * Run GET operation on all cells and merge the result.
+     */
+    public static <T> Either<List<T>, WebApplicationException> doGetAndMerge(CellWebClientConnector cellWebClientsSupplier,
+                                                                             String path,
+                                                                             ParameterizedTypeReference<List<T>> type,
+                                                                             long timeoutMs) {
+        Either<List<List<T>>, Throwable> partials;
+        try {
+            partials = Flux
+                    .merge(
+                            cellWebClientsSupplier.getWebClients().values().stream()
+                                    .map(c ->
+                                            c.get().uri(path)
+                                                    .retrieve()
+                                                    .bodyToMono(type)
+                                    )
+                                    .collect(Collectors.toList())
+                    )
+                    .collectList()
+                    .<Either<List<List<T>>, Throwable>>map(Either::ofValue)
+                    .onErrorResume(e -> Mono.just(Either.ofError(e)))
+                    .block(Duration.ofMillis(timeoutMs));
+        } catch (Exception e) {
+            return Either.ofError(new WebApplicationException(e, Response.Status.INTERNAL_SERVER_ERROR));
+        }
+
+        if (partials == null) {
+            return Either.ofError(new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR));
```

Comment:
minor: add an error message (e.g. `No results from any cell`) to aid debugging in the future

Reply:
I will add some log lines.